src/kOS.Safe/UpdateHandler.cs

```diff
@@ -9,27 +8,22 @@ public class UpdateHandler
         // insert itself more than once into the observer list, it still only gets in the list
         // once and therefore only gets its Update() called once per update.
         // The value of the KeyValuePair, the int, is unused.
-        private readonly Dictionary<IUpdateObserver, int> observers = new Dictionary<IUpdateObserver, int>();
-        
+        private readonly HashSet<IUpdateObserver> observers = new HashSet<IUpdateObserver>();
+        private readonly HashSet<IUpdateObserver> fixedObservers = new HashSet<IUpdateObserver>();
```

Comment:
I was searching the .net docs for something like a hash and could't find it and gave up and used Dictionary.